File: Recursion-with-Arrays/cpp/Max_of_an_Array.cpp
#include <iostream>
using namespace std;

int max(int arr[], int idx, int n){
    // write your code here
    if(idx==n){
        return -1;
    }
    int mx=max(arr,idx+1,n);
    if(arr[idx]>mx){
        mx=arr[idx];
    }
    return mx;
    
}

int main(){
    int n;
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++)
        cin >> arr[i];
    int p = max(arr, 0, n);
    cout << p << endl;
}

File: Level-1/7. Recursion-with-Arrays/java/Max_of_an_Array.java
import java.io.*;
import java.util.*;

public class Main {

    public static void main(String[] args) throws Exception {
        // write your code here
        Scanner scn = new Scanner(System.in);

        int n = scn.nextInt();

        int arr[] = new int[n];
        for (int i = 0; i < n; i++) {
            arr[i] = scn.nextInt();
        }
        // int max=Integer.MIN_VALUE;
        System.out.println(maxOfArray(arr,0));
    }

    public static int maxOfArray(int[] arr, int idx) {
        if(idx==arr.length){
            return 0;
        }
        int newmx=maxOfArray(arr,idx+1);
        if(newmx<arr[idx]){
            newmx=arr[idx];
        }
        return newmx;
    }

}